### ISSUE TITLE
[Notebook QA] Notebook doesn't open the selected Group if localStorage is unavailable

### ISSUE THREAD

**Issue by mkdir-washington-edu** [hypothesis/client#3251]
In MS Edge, when you select a new Hypothesis Sections group in the Sidebar and open Notebook the available Section in Notebook is the first available Section in the picker instead of the selected section.

**To Reproduce**
1. Go to https://hypothesis.instructure.com/courses/237/assignments/1194
2. Change the Section to "Reading group 2"
3. Open Notebook
4. Note the selected section is "2020FA - POLYMER CHEMIST..."

**Expected behavior**
In other browsers the Notebook section is whatever Section was selected when Notebook opens.

**Desktop (please complete the following information):**
 - OS: Windows 10 Home v 1909
 - Browser: MS Edge 89.0

**Comment by LMS007** [comment]
@mkdir-washington-edu 

Ugh...I'm not able to repro this with
Microsoft Edge
Version 89.0.774.68 (Official build) (64-bit)
On Win10 Pro

Lets try and sync up tomorrow

![Screenshot_2](https://user-images.githubusercontent.com/3939074/113639289-a9b77780-962d-11eb-85ab-cd01fb48e996.png)

**Comment by mkdir-washington-edu** [comment]
@LMS007 I repro'd the error in a video. https://drive.google.com/file/d/1pmg8oJFej_zWEEet6Om2yUYHw9HxZgA8/view?usp=sharing

**Comment by mkdir-washington-edu** [comment]
I repro'd this issue when blocking cookies from hypothes.is, lms.hypothes.is, and via4.hypothes.is. I can not repro the issue if I allow cookies from those URLs. If we decide we expect users to allow cookies from those domains we should make sure that information is very upfront in our eventual Notebook KB articles.

When blocking cookies I get the following console errors:
![edge](https://user-images.githubusercontent.com/22603275/113779334-907cfc80-96fb-11eb-96c0-b744d2939a43.PNG)

Also, when Kyle opened the Sidebar's account menu using the eng+canvasteacher account, he had both the LMS display name and access to Notebook. When I opened the Sidebar's account menu using jeremydean+teacher, I only got the LMS display name. Unsure if this is intentional.

**Comment by LMS007** [comment]
I confirmed this happens in Chrome too. (makes sense)

This happened on loading up the assignment in edge / chrome (probably other browser too) when you disable third party cookies  irrespective of the notebook.  The error happened when we try and write pdfjs.preferences to local storage. `localStorage.getItem('pdfjs.preferences')));`


![Screenshot_3](https://user-images.githubusercontent.com/3939074/113791562-cec4ec80-96f8-11eb-83ae-3241275fbbab.png)

**Comment by lyzadanger** [comment]
The problem behavior here can be reproduced by:

* Blocking 3rd-party cookies in your browser (which also affects `localStorage` availability)
* Viewing an assignment with sections enabled in Canvas
* Selecting a group other than the first section
* Launching the Notebook

The bad behavior is that the Notebook will always load annotation activity for the first group in the list of available section groups, not the group that is selected in the sidebar.

This behavior results from a combination of several factors:

Most vitally: **configuration settings encoded in the URL, which include the group ID that the Notebook should load annotations for, are ignored in the LMS context because configuration is instead provided (exclusively) by configuration objects exchanged via inter-iframe messaging.** This is the crux of the issue: we’re not able to get configuration from the URL within LMS contexts.

This URL-config-ignoring behavior means that the application is never able to populate the “direct-linked group ID” from the URL as it is intended to. Because the direct-linked group ID value ends up as `null`, the code has a fallback to load up annotations for the currently-focused group—as stored in `localStorage`. This `localStorage` fallback is why the Notebook appears to work correctly (load annotations for the group selected in the sidebar) in other LMS cases.

The reason it loads annotations in the first available group in the problem-behavior case at all (versus not loading anything) is that there is yet another wrinkle! There is a reducer in the store that is triggered after the list of available groups is fetched. This reducer looks at the current value of the `focusedGroupId`, and if it is `null`, it will (in this case), set it to the `id` of the first group in the list of loaded groups. The `NotebookView` sees this change of `focusedGroupId`et voila! It loads the annotations for that group—the first group in the list.

To summarize: The **main problem here is that we’re unable currently to pass configuration via URL parameters when in the LMS context**. This is papered over by the `localStorage` fallbacks and other details, but this configuration-source issue needs to be solved.
